kuma/wiki/models.py

```diff
@@ -1519,6 +1519,13 @@ def clean_current_revision(self, user):
                 rev.based_on_id = prior_pk
             rev.comment = 'Clean prior revision of {} by {}'.format(
                 prior_created, prior_creator)
+            # The current revision sometimes has an old slug that's
+            # different than its document's current slug, so let's ensure
+            # that they're the same so we don't trigger unique index errors
+            # when the document is saved within the make_current() call made
+            # within the rev.save() call below, or explicitly further below.
+            rev.slug = self.slug
+            rev.title = self.title
```

Comment:
I don't know enough (or really anything) about our DB schemas. Based on this code, I'm assuming that the document slug and title are the source of truth, and those fields on the revisions aren't definitive. And that there must be or must have been a bug where when a writer changes a slug or title we updated the document but didn't update (or create a new) current revision.  If that is basically correct, then this makes sense to me.

Reply:
Yes, that's my understanding as well.